s3.tf

```diff
@@ -89,26 +93,43 @@ resource "aws_s3_bucket_inventory" "apiary_bucket" {
 }
 
 resource "aws_s3_bucket_public_access_block" "apiary_bucket" {
-  count  = var.s3_block_public_access == true ? "${length(local.apiary_data_buckets)}" : 0
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.s3_block_public_access == true ? {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema
+  } : {}
+  bucket = aws_s3_bucket.apiary_data_bucket[each.key].id
 
   block_public_acls   = true
   block_public_policy = true
   ignore_public_acls  = true
 }
 
 resource "aws_s3_bucket_notification" "data_events" {
-  count  = "${var.enable_data_events == "" ? 0 : length(local.apiary_data_buckets)}"
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.enable_data_events == "" ? {} : {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema if lookup(schema, "enable_data_event_queue", "0") == "0"
+  }
+  bucket = aws_s3_bucket.apiary_data_bucket[each.key].id
 
   topic {
-    topic_arn = "${aws_sns_topic.apiary_data_events.*.arn[count.index]}"
+    topic_arn = aws_sns_topic.apiary_data_events[each.key].arn
     events    = ["s3:ObjectCreated:*", "s3:ObjectRemoved:*"]
   }
 }
 
+resource "aws_s3_bucket_notification" "data_queue_events" {
+  for_each = { for schema in local.schemas_info : "${schema["schema_name"]}" => schema if lookup(schema, "enable_data_event_queue", "0") == "1" }
```

Comment:
can we use `enable_data_events` and bool values `true/false`

Reply:
No, we can't.  Remember from our per-schema `tags` investigation that every key/value in the `apiary_managed_schemas` map *MUST* be a string, otherwise it doesn't work unless every possible key/value is specifically set by the caller.  Best we could do is to use "true" and "false" strings and then call `tobool` on them to cast.  Not sure that's any better, though, and may be also confusing.

Reply:
ok makesense.